src/test/java/com/browserstack/steps/SetupSteps.java

```diff
@@ -0,0 +1,141 @@
+package com.browserstack.steps;
+
+import com.browserstack.ParallelTest;
+import com.browserstack.local.Local;
+import com.browserstack.util.OsUtility;
+import com.browserstack.util.Utility;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.Scenario;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.io.FileReader;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class SetupSteps {
+
+    private final StepData stepData;
+    private Local bstackLocal;
+    protected String driverBaseLocation = Path.of(System.getProperty("user.dir"), "/src/test/resources/drivers").toString();
+
+    private static final String PASSED = "passed";
+    private static final String FAILED = "failed";
+    private static final String URL = "https://bstackdemo.com";
+    private static final String WEBDRIVER_CHROME_DRIVER = "webdriver.chrome.driver";
+    private static final String DOCKER_SELENIUM_HUB_URL = "http://localhost:4444/wd/hub";
+    private static final String BROWSERSTACK_HUB_URL = "https://hub.browserstack.com/wd/hub";
+    private static final String CAPABILITY_CONFIG_FILE = "src/test/resources/config/caps.json";
+    private static final String REPO_NAME = "browserstack-examples-cucumber-junit5 - ";
+
+    public SetupSteps(StepData stepData) {
+        this.stepData = stepData;
+    }
+
+    @Before
+    public void setUp(Scenario scenario) throws Exception {
```

Comment:
Can we break this method into smaller method?

Reply:
refactored into smaller functions